cogs/tags/tags.py

```diff
@@ -0,0 +1,1024 @@
+
+        tags.sort()
+
+        if tags:
+            try:
+                self.dm = self.settings.get("dm", False)
+                if self.dm:
+                    await ctx.send("Check your DMs.")
+                    msg = "Here are a list of tags for {}:\n```".format(ctx.message.author.mention)
+                    for item in tags:
+                        if len(msg) + len(item) > 1990:
+                            msg += "```"
+                            await ctx.author.send(msg)
+                            msg = "```"
+                        msg += item + "\n"
+                    msg += "```"
+                    await ctx.author.send(msg)
+                else:
+                    p = Pages(ctx=ctx, entries=tags, show_entry_count=True)
+                    p.embed.colour = 0x738bd7 # blurple
+                    p.embed.set_author(name=owner.display_name, icon_url=owner.avatar_url or
+                                       owner.default_avatar_url)
+                    await p.paginate()
+            except Exception as e:
+                await ctx.send(e)
+        else:
+            await ctx.send('{0.name} has no tags.'.format(owner))
+
+    @tag.command(name="all")
+    @commands.guild_only()
+    async def _all(self, ctx):
+        """Lists all server-specific tags for this server."""
+
+        tags = [tag.name for tag in self.config.get(str(ctx.message.guild.id), {}).values()]
+        tags.sort()
+
+        if tags:
+            try:
+                self.dm = self.settings.get("dm", False)
+                if self.dm:
+                    await ctx.send("Check your DMs.")
+                    msg = "Here are a list of tags for {}:\n```".format(ctx.message.guild.name)
+                    for item in tags:
+                        if len(msg) + len(item) > 1990:
+                            msg += "```"
+                            await ctx.author.send(msg)
+                            msg = "```"
+                        msg += item + "\n"
+                    msg += "```"
+                    await ctx.author.send(msg)
+                else:
+                    p = Pages(ctx=ctx, entries=tags, per_page=15, show_entry_count=True)
+                    p.embed.colour =  0x738bd7 # blurple
+                    await p.paginate()
+            except Exception as e:
+                await ctx.send(e)
+        else:
+            await ctx.send('This server has no server-specific tags.')
+
+    @tag.command(name="purge")
+    @commands.guild_only()
+    @checks.mod_or_permissions(manage_messages=True)
+    async def purge(self, ctx: Context, member: discord.Member):
+        """Removes all server-specific tags by a user.
+        You must have Manage Messages permissions to use this.
+        """
+        db = self.config.get(str(ctx.message.guild.id), {})
+        tags = [key for key, tag in db.items() if tag.owner_id == str(member.id)]
+
+        # TODO I'm pretty sure there's a decorator for the following.
```

Comment:
Is this still a todo?

Reply:
Yeah gonna leave this one a todo since this works as before but we can probably eliminate this later.